### PR TITLE
Seperate namespaces for V1 and V2 controller paths

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?
I'm working on some new automation to test cgroups code on different platforms + containers, and this was the first bug I noticed. On hybrid systems, the code that fetches the Controller paths can overwrite the `cpu` controller on V1 if it exists on V2, as it shares a name with the V1 controller. That's bad.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

- build and run on a linux system running hybrid cgroups (for example, Ubuntu 18.04)
- Run `system/process`, ensure that the `cpu` fields in `cgroup` is populated with valid data

### CODE REVIEW

File: libbeat/metric/system/cgroup/util_test.go [hunk 1]
```diff
@@ -171,17 +171,17 @@ func TestProcessCgroupPaths(t *testing.T) {
...
-	assert.Equal(t, path, paths["cpuacct"].ControllerPath)
-	assert.Equal(t, path, paths["cpuset"].ControllerPath)
-	assert.Equal(t, path, paths["devices"].ControllerPath)
-	assert.Equal(t, path, paths["freezer"].ControllerPath)
-	assert.Equal(t, path, paths["memory"].ControllerPath)
-	assert.Equal(t, path, paths["net_cls"].ControllerPath)
-	assert.Equal(t, path, paths["net_prio"].ControllerPath)
-	assert.Equal(t, path, paths["perf_event"].ControllerPath)
-	assert.Len(t, paths, 10)
+	assert.Equal(t, path, paths.V1["blkio"].ControllerPath)
```
Comment (ChrsMark) [inline libbeat/metric/system/cgroup/util_test.go:174]: Wondering if we could replicate the "buggy" case with a test somehow 🤔

**Review by ChrsMark — APPROVED** [review]
lgtm! 
Since it is a bugfix, how about adding a changelog too?

note: I'm not super familiar with this codebase but the path itself looks good so feel free to merge it if CI is green and you are confident with the change.

**Review by MichaelKatsoulis — APPROVED** [review]
LGTM!

File: libbeat/metric/system/cgroup/util.go [hunk 2]
```diff
@@ -60,6 +60,26 @@ type ControllerPath struct {
...
 }
 
+// PathList contains the V1 and V2 controller paths in a process
+// Separate the V1 and V2 cgroups so we don't have hybrid cgroups fighting for one namespace
+type PathList struct {
+	V1 map[string]ControllerPath
+	V2 map[string]ControllerPath
+}
+
+// Flatten combines the V1 and V2 cgroups in cases where we don't need a map with keys
```
[the remark this answers is not in the document]
Comment (fearful-symmetry) [reply libbeat/metric/system/cgroup/util.go:70]: So, the separation is for cases where we actually need a map, i.e, the cgroups code that reports the metrics by-name. A lot of uses cases will just blindly literate over the controller filepaths (usually looking for a docker container name, or something like that) and they don't care what the names of the cgroups are.

**Pull request merged**